### PR TITLE
Introduced explicit control of filters in the NZS mode

### PR DESCRIPTION
**Pull request by igv4321**
Per discussion with NZS data consumers (Olga Kodolova), most HBHE filters are turned off in the Phase 1 NZS mode, and reco parameters are taken from DB

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoLocalCalo/Configuration/python/hcalLocalRecoNZS_cff.py [hunk 1]
```diff
@@ -22,11 +22,18 @@
...
 _phase1_hbherecoMB = RecoLocalCalo.HcalRecProducers.HBHEPhase1Reconstructor_cfi.hbheprereco.clone(
     dropZSmarkedPassed = cms.bool(False),
-    recoParamsFromDB = cms.bool(False),
     algorithm = dict(
         useM2 = cms.bool(False),
         useM3 = cms.bool(False)
     ),
+    setNegativeFlagsQIE8 = cms.bool(False),
+    setNegativeFlagsQIE11 = cms.bool(False),
+    setNoiseFlagsQIE8 = cms.bool(True),
```
Comment (perrotta) [inline RecoLocalCalo/Configuration/python/hcalLocalRecoNZS_cff.py]: Just to check (as I'm not sure I fully understood the logic in the HBHEPhase1Reconstructor): shouldn't this setNoiseFlagsQIE8 be set to "False", as the other bools here, if you want to retrieve the info from DB (and therefore not to look it up in the constructor) ?

Comment (igv4321) [inline RecoLocalCalo/Configuration/python/hcalLocalRecoNZS_cff.py]: "setNoiseFlagsQIE8" switch turns on certain filters in the rechit processing. These filters, in turn, set certain rechit flags. It looks like for NZS mode only this particular filter can provide some useful information (at least this is what Olga told me), while the parameters of this filter have not been derived yet for QIE11. 

The use of filters is unrelated to choosing reco algorithm parameters from db or from cfi. This is an orthogonal issue.

**Comment by perrotta** [comment]
+1
Following the comment https://github.com/cms-sw/cmssw/pull/18100#discussion_r109055639  (thank you Igor), I see no otherpossible  issues in the implementation.
Jenkins tests show no significant changes, as they should

**Comment by davidlange6** [comment]
+1

**Pull request merged**